modyn/config/schema/pipeline/trigger/cost/cost.py

```diff
@@ -0,0 +1,74 @@
+"""Cost aware triggers are evaluate the trade-off between the cost of a trigger
+(e.g. measured by training machine time) and the benefit of the trigger (e.g.
+measured by a regret metric like data incorporation latency or avoidable
+misclassification incorporation latency)."""
+
+from typing import Literal
+
+from pydantic import Field
+
+from modyn.config.schema.base_model import ModynBaseModel
+
+from ..performance.performance import _InternalPerformanceTriggerConfig
+
+
+class _CostTriggerConfig(ModynBaseModel):
+    """Base class for cost aware trigger policies."""
+
+    cost_tracking_window_size: int = Field(
+        description="How many trigger into the past should be considered for the linear cost w.r.t. data amount model."
+    )
+
+
+class DataIncorporationLatencyCostTriggerConfig(_CostTriggerConfig):
+    """Cost aware trigger policy configuration that uses the data incorporation
+    latency as a regret metric.
+
+    While no trigger occurs the un-integrated data is cumulated and the sum of all seconds of every unincorporated data
```

Comment:
I don't get that sentence. I understand you cumulate the latencies, but how do you also cumulate the data? Or is this redundant?

Reply:
From the samples, we can build an unintegrated samples over a time curve.
The area under this curve is the (cumulated) latency.

Reply:
Okay, since we're discrete I am just thinking of it as a summation over all individual latencies, but I am not sure whether this is true.

Reply:
Yes, instead a computationally and memory-wise more expensive alternative would be to store all samples since the last trigger and on every evaluation compute the time distance (latency) to the current time and build the sum of these latencies. The approach we take here is iterative and doesn't require remembering the timestamps of the elements.